### ISSUE TITLE
Dropdown: setsize not updated when options are added

### ISSUE THREAD

**Issue by ahackettms** [microsoft/fluentui#14375]
### Environment Information

- **Package version(s)**: 7.124.0
- **Browser and OS versions**: not relevant

### Please provide a reproduction of the bug in a codepen:

Codepen repro: https://codepen.io/ahackettms/pen/ZEWEmWj

#### Actual behavior:
If Dropdown options are added after initial rendering, the dropdown setsize is not updated, causing incorrect information when using Narrator to read the dropdown information.
Also the dropdown items have incorrect position information. 

#### Expected behavior:
The Dropdown setsize should be updated when options are updated. 
The issue is that in UNSAFE_componentWillReceiveProps(), newProps.options is the same as  this.props.options , so this._sizePosCache.updateOptions is not called. 

### Priorities and help requested:

Are you willing to submit a PR to fix? ( No)

Requested priority: High, it's expected to fix by end of August for accessibility requirements for Microsoft Lists app

Products/sites affected: SharePoint Lists/STS. This affects accessibility compliance for the Lists app

**Comment by khmakoto** [comment]
Hi @ahackettms, I can confirm that this is happening and will work on a fix as soon as possible!
